File: plugins/commands/serve/client/state_bag.rb
# Copyright (c) HashiCorp, Inc.
# SPDX-License-Identifier: MIT

module VagrantPlugins
  module CommandServe
    class Client
      class StateBag < Client
        # @param [String]
        # @return [String]
        def get(key)
          req = SDK::StateBag::GetRequest.new(
            key: key
          )
          client.get(req).value
        end
        alias_method :[], :get

        # @param [String]
        # @return [String, Boolean]
        def get_ok(key)
          req = SDK::StateBag::GetRequest.new(
            key: key
          )
          resp = client.get_ok(req)
          return resp.value, resp.ok
        end

        # @param [String, String]
        # @return []
        def put(key, val)
          req = SDK::StateBag::PutRequest.new(
            key: key, value: val
          )
          client.put(req)
        end
        alias_method :[]=, :put

        # @param [String]
        # @return []
        def remove(key)
          req = SDK::StateBag::RemoveRequest.new(
            key: key
          )
          client.remove(req)
        end
        alias_method :delete, :remove

      end
    end
  end
end

File: internal/core/state_bag.go
// Copyright (c) HashiCorp, Inc.
// SPDX-License-Identifier: MIT

package core

import (
	"github.com/hashicorp/vagrant-plugin-sdk/core"
)

// The StateBag keeps the state of Vagrant during execution
type StateBag struct {
	state map[string]interface{}
}

func NewStateBag() *StateBag {
	return &StateBag{
		state: map[string]interface{}{},
	}
}

// Get implements core.StateBag
func (s *StateBag) Get(key string) interface{} {
	return s.state[key]
}

// GetOk implements core.StateBag
func (s *StateBag) GetOk(key string) (val interface{}, ok bool) {
	val, ok = s.state[key]
	return
}

// Put implements core.StateBag
func (s *StateBag) Put(key string, val interface{}) {
	s.state[key] = val
}

// Remove implements core.StateBag
func (s *StateBag) Remove(key string) {
	delete(s.state, key)
}

var _ core.StateBag = (*StateBag)(nil)
